src/hotspot/share/opto/phaseX.cpp

```diff
@@ -1955,6 +1959,21 @@ Node *PhaseCCP::transform( Node *n ) {
   GrowableArray <Node *> trstack(C->live_nodes() >> 1);
 
   trstack.push(new_node);           // Process children of cloned node
+
+  // This CCP pass may prove that no exit test for a loop ever succeeds (i.e. the loop is infinite). In that case,
+  // the logic below doesn't follow any path from Root to the loop body (they are proven never taken). If CCP only works
```

Comment:
May be clarify that "any path from Root's inputs". Because any such loop is reachable from Root's uses/outputs.

Reply:
I reworded the comment. Let me know if that looks good to you.